src/eei.h

```diff
@@ -79,6 +79,109 @@ struct EthereumInterface : ShellExternalInterface {
   void memoryCopy(std::vector<char> const& src, uint32_t srcoffset, uint32_t dstoffset, uint32_t length);
   void copyAddressToMemory(struct evm_address const& address, uint32_t dstoffset);
 
+  /*
+   * Memory Operations
+   */
+  
+  void loadMemory(uint32_t srcOffset, uint8_t *dst, size_t length)
+  {
+      assert(dst != nullptr);
+      assert(length);
+      
+      for (uint32_t i = 0; i < length; ++i) {
+          *(dst + i) = memory.get<uint8_t>(srcOffset + i);
+      }
+  }
+
+  void storeMemory(uint32_t dstOffset, const uint8_t *src, size_t length)
+  {
+      assert(src != nullptr);
+      assert(length);
+
+      for (uint32_t i = 0; i < length; ++i) {
+          memory.set<uint8_t>(dstOffset + i, *(src + i));
+      }
+  }
+
+  /*
+   * Memory Op Wrapper Functions
+   */
+
+  void loadUint256(uint32_t srcOffset, struct evm_uint256be *dst)
+  {
+      assert(dst != nullptr);
+      loadMemory(srcOffset, dst->bytes, 32);
+      endianSwap(dst->bytes, 32);
+  }
+
+  void storeUint256(uint32_t dstOffset, struct evm_uint256be *src)
+  {
+      assert(src != nullptr);
+      endianSwap(src->bytes, 32);
+      storeMemory(dstOffset, src->bytes, 32);
+  }
+
+  void loadUint160(uint32_t srcOffset, struct evm_address *dst)
+  {
+      assert(dst != nullptr);
+      loadMemory(srcOffset, dst->bytes, 20);
+      endianSwap(dst->bytes, 20);
+  }
+
+  void storeUint160(uint32_t dstOffset, struct evm_address *src)
+  {
+      assert(src != nullptr);
+      endianSwap(src->bytes, 20);
+      storeMemory(dstOffset, src->bytes, 20);
+  }
+
+  void loadUint128(uint32_t srcOffset, struct evm_uint256be *dst)
+  {
+      assert(dst != nullptr);
+      loadMemory(srcOffset, dst->bytes, 16);
+      endianSwap(dst->bytes, 32);
+  }
+
+  void storeUint128(uint32_t dstOffset, struct evm_uint256be *src)
+  {
+      assert(src != nullptr);
+      assert(!exceedsUint128(src));
+      endianSwap(src->bytes, 32);
+      storeMemory(dstOffset, src->bytes, 16);
+  }
+
+  /*
+   * Utilities
+   */
+  
+  /* Checks if host supplied 256 bit big-endian value exceeds UINT128_MAX */
+  unsigned int exceedsUint128(const struct evm_uint256be *value)
```

Comment:
In general, in C++ (otherwise than in C) you can skip the `struct` keyword. Also for struct arguments pass them by reference `const evm_uint256be& value`. You will not have to check if the `value != nullptr`.

Reply:
The helper functions already have these changes in #19 .